demo-apps/chaos-monkey-demo-app-unleash-toggles/src/main/java/com/example/chaos/monkey/toggledemo/ChaosDemoApplication.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.chaos.monkey.toggledemo;
+
+import de.codecentric.spring.boot.chaos.monkey.configuration.ChaosMonkeyProperties;
+import de.codecentric.spring.boot.chaos.monkey.configuration.toggles.ChaosToggleNameMapper;
+import no.finn.unleash.Unleash;
+import no.finn.unleash.UnleashContext;
+import no.finn.unleash.UnleashContextProvider;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+
+@SpringBootApplication
+public class ChaosDemoApplication {
+
+  public static void main(String[] args) {
+    SpringApplication.run(ChaosDemoApplication.class, args);
+  }
+
+  @Bean
+  public UnleashContextProvider unleashContextProvider() {
+    return () -> {
+      UnleashContext.Builder context = new UnleashContext.Builder();
+
+      Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+      if (auth != null && auth.getPrincipal() instanceof User) {
+        context.userId(((User) auth.getPrincipal()).getUsername());
+      }
+
+      return context.build();
+    };
+  }
+
+  @Bean
+  public Unleash unleash(UnleashContextProvider contextProvider) {
+    UserAwareFakeUnleash unleash = new UserAwareFakeUnleash(contextProvider);
+
+    //        unleash.enable("chaos.monkey.controller");
```

Comment:
```suggestion

```

Reply:
I left this comment as a way to demonstrate how someone could toggle a feature without having to run an actual Unleash server. Should I remove it or just add additional comments explaining its use?